Documentation/ExtensionConfiguration/Index.rst

```diff
@@ -15,35 +15,40 @@ advise) the extension needs to contain its own :file:`composer.json`.
 
 .. include:: /CodeSnippets/ExtensionConfiguration/ComposerJson.rst.txt
 
-For historic reasons TYPO3 extension names have to be written in lower case
-separated by underscores. We suggest to use the extension key for the directory
-of the extension as well to minimize confusion. So the extension in the path
-:file:`site_package` has to have the
-same "extension-key" to be defined in the "extra" section of the composer.json.
+For historic reasons TYPO3 extension names are written in
+lower case words and separated by underscores if there are more than one. This
+is known as the extension key.The directory containing the extension should have
+the same name as the extension key. Composer package names are written in
+lower-case words but are separated with dashes if there is more than one word.
+
+At the top of the composer.json file we see the composer package name
+:file:`t3docs/site-package` (with a dash) and at the bottom we see the TYPO3
+extension key in the extras section - :file:`site_package` (with an underscore).
+The Composer "name" consists of a vendor name followed by a forward slash and the
+lowercase extension name with dashes.
```

Comment:
It does not have to be the extension key with dashes. Actually, it can be any name and may have nothing in common with the extension key. I am doing this in my extensions, for example, "matomo_widgets" as extension key, and "typo3-matomo-widgets" as part of the Composer package name.

If we let this as is, please adjust "extension name" to "extension key"

Reply:
Hi Chris I found this really confusing in the documentation when I tried to follow it. I only wanted it to work and so the simplest thing for me was just to have site-package and site_package. Anything else was just confusing at this point (for me)